doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -278,6 +278,19 @@ At this time, we do not make any guarantees about the freshness of these numbers
 
 <!-- RELATION_SPEC_UNDOCUMENTED mz_internal.mz_global_frontiers -->
 
+### `mz_hydration_statuses`
+
+The `mz_hydration_statuses` view describes the per-replica hydration status of each object maintained by a dataflow.
+
+A dataflow-maintained object is hydrated on a given replica when dataflow has fully processed the initial snapshot of data available in its inputs.
+
+<!-- RELATION_SPEC mz_internal.mz_hydration_statuses -->
+| Field        | Type        | Meaning                                                                                                                                                                                                                                                                                                                                                          |
+| -----------  | ----------- | --------                                                                                                                                                                                                                                                                                                                                                         |
+| `object_id`  | [`text`]    | The ID of a dataflow-maintained object. Corresponds to [`mz_catalog.mz_indexes.id`](../mz_catalog#mz_indexes), [`mz_catalog.mz_materialized_views.id`](../mz_catalog#mz_materialized_views), [`mz_internal.mz_subscriptions`](#mz_subscriptions), [`mz_catalog.mz_sources.id`](../mz_catalog#mz_sources), or [`mz_catalog.mz_sinks.id`](../mz_catalog#mz_sinks). |
+| `replica_id` | [`text`]    | The ID of a cluster replica.                                                                                                                                                                                                                                                                                                                                     |
```

Comment:
Last we talked about this, I thought we weren't going to make this per replica, so that it would be easy to see if a dataflow was stalled because there are no replicas? I guess I would have to left join in this case, and take the null to mean there are no replicas?

Reply:
Ah, back then we were still talking about a more complicated general status view for objects. This one I envisioned more as an extension of `mz_compute_hydration_statuses` to also storage objects.

Left joining will work, although you'd have to be careful what you join against. Some source types don't appear here because they don't run a dataflow, so if you simply joined against `mz_sources` that might lead you to believe that, e.g., a webhook source is not hydrated because it doesn't have a replica, while in actuality these sources don't run on replicas and don't hydrate in the first place.

I'm happy to remove the `replica_id` column and make this view a global one if this is more usable. If we end up needing it, we can always add an `mz_cluster_replica_hydration_statuses` later. Note that an aggregated view would show `false` when no replica exists, same as when a replica exists but has not finished hydrating the dataflow. If you'd want to find out which of the two it is, you'd still need to query the cluster replicas separately.

Reply:
The webhook case is really confusing :( Is there a way we include webhook sources and just always call them hydrated?

Reply:
Thinking more about this, it seems useful to expose the difference between no replicas and not yet hydrated, so we can keep the replica column.